universal-application-tool-0.0.1/app/repository/DbRepository.java

```diff
@@ -0,0 +1,107 @@
+package repository;
+
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import io.ebean.Ebean;
+import io.ebean.EbeanServer;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Applicant;
+import models.Program;
+import models.Question;
+import play.db.ebean.EbeanConfig;
+
+public class DbRepository {
```

Comment:
Style question kind of for the future: Will all of our queries go in this class / is that a best practice? Or if we started to have more complex and many queries would it potentially make sense to break these out into separate classes?

Reply:
Hm, are you sure?  I don't like the aesthetics of that - really spreads boilerplate around.  It's been a minute since my last java job but that feels messy to me.

Reply:
A repo per entity feels better in terms of single-responsibility imo. To take care of the boilerplate you could have a package-private superclass:

```java
class DbRepository {
  private final EbeanServer ebeanServer;
  private final DatabaseExecutionContext executionContext;

  @Inject
  public DbRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext executionContext) {
    this.ebeanServer = Ebean.getServer(ebeanConfig.defaultServer());
    this.executionContext = executionContext;
  }
}


public class ProgramRepository extends DbRepository {
  ...
}
```